plugins/extra/runescape.go

```diff
@@ -118,115 +120,175 @@ func getCombatLevel(attack, defence, strength, hitpoints, ranged, prayer, magic
-	var emptySkill runescapeLevelMetadata
+func (p *runescapePlugin) getPlayerSkills(search string) (map[string]runescapeLevelMetadata, error) {
+	var emptySkills map[string]runescapeLevelMetadata
 
 	found := false
 	player := ""
-	skill := ""
+	skillsString := ""
+	var skills []string
 
 	matches := levelRegex.FindAllStringSubmatch(search, -1)
 	for _, v := range matches {
 		if strings.HasPrefix(v[1], "\"") {
 			v[1] = v[1][1 : len(v[1])-1]
 		}
 		player = v[1]
-		skill = v[2]
+		skillsString = v[2]
+		skills = whitespaceRegex.Split(skillsString, -1)
 		found = true
 	}
 
 	if !found {
-		return emptySkill, errors.New("Unable to parse player or skill")
+		return emptySkills, errors.New("Unable to parse player or skill")
 	}
 
 	resp, err := http.Get("https://secure.runescape.com/m=hiscore_oldschool/index_lite.ws?player=" + player)
 	if err != nil {
-		return emptySkill, err
+		return emptySkills, err
 	}
 
 	bytes, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return emptySkill, err
+		return emptySkills, err
 	}
 	data := strings.Split(strings.TrimSpace(string(bytes)), "\n")
 
 	// It's not strictly needed to build all this up, but it may be useful later.
-	var ret = make(map[string]runescapeLevelMetadata)
+	var ret = make(map[string]runescapeLevelMetadata, len(skills))
 	if len(data) < len(runescapeOldSchoolSkillNames) {
-		return emptySkill, fmt.Errorf("Invalid data")
+		return emptySkills, fmt.Errorf("Invalid data")
 	}
 
 	for i, name := range runescapeOldSchoolSkillNames {
 		md, err := newRunescapeLevelMetadata(name, player, data[i])
 		if err != nil {
-			return emptySkill, err
+			return emptySkills, err
 		}
 
 		ret[md.Skill] = md
 	}
 
-	if skill == "combat" {
-		combat := getCombatLevel(
-			ret["attack"].Level,
-			ret["defence"].Level,
-			ret["strength"].Level,
-			ret["hitpoints"].Level,
-			ret["ranged"].Level,
-			ret["prayer"].Level,
-			ret["magic"].Level)
-		return runescapeLevelMetadata{
-			Rank:   -1,
-			Level:  combat,
-			Exp:    -1,
-			Player: player,
-			Skill:  skill,
-		}, nil
-	}
+	returnedSkills := make(map[string]runescapeLevelMetadata)
+
+	for _, skill := range skills {
+		if skill == "combat" {
+			combat := getCombatLevel(
+				ret["attack"].Level,
+				ret["defence"].Level,
+				ret["strength"].Level,
+				ret["hitpoints"].Level,
+				ret["ranged"].Level,
+				ret["prayer"].Level,
+				ret["magic"].Level)
+			returnedSkills["combat"] = runescapeLevelMetadata{
+				Rank:   -1,
+				Level:  combat,
+				Exp:    -1,
+				Player: player,
+				Skill:  skill,
+			}
+			continue
+		}
+
+		// Pull out the proper data
+		md, ok := ret[skill]
+		if !ok {
+			return emptySkills, fmt.Errorf("Unknown skill %q", skill)
+		}
 
-	// Pull out the proper data
-	md, ok := ret[skill]
-	if !ok {
-		return emptySkill, fmt.Errorf("Unknown skill %q", skill)
+		returnedSkills[skill] = md
 	}
 
-	return md, nil
+	return returnedSkills, nil
+}
+
+func sortedSkillNames(skills *map[string]runescapeLevelMetadata) []string {
```

Comment:
Maps are passed by reference so you shouldn't need a pointer here

Reply:
:+1: